src/main/java/me/jellysquid/mods/lithium/mixin/world/layer_caching/MixinParentedLayer.java

```diff
@@ -0,0 +1,34 @@
+package me.jellysquid.mods.lithium.mixin.world.layer_caching;
+
+import me.jellysquid.mods.lithium.common.world.layer.LayerFactoryUtil;
+import net.minecraft.world.biome.layer.*;
+import net.minecraft.world.biome.layer.type.ParentedLayer;
+import net.minecraft.world.biome.layer.util.LayerFactory;
+import net.minecraft.world.biome.layer.util.LayerSampleContext;
+import net.minecraft.world.biome.layer.util.LayerSampler;
+import org.spongepowered.asm.mixin.Mixin;
+
+/**
+ * Memoize the LayerFactory created by the ParentedLayer such that layers don't get duplicated when accessed from
+ * multiple children layers
+ *
+ * Because we can't apply mixin to default method, just override in classes that get duplicated
```

Comment:
I'm not too thrilled about needing to override a default method in multiple classes.

It looks like Mixins is breaking down when trying to overwrite methods in the interface due to the fact that lambdas are being used in a rather complex way. It is possible to change this class to an interface and then overwrite the defaulted method by moving this code into a static helper class elsewhere, which would in turn allow the JVM to compile this.